### PR TITLE
[SRVKE-1193] Remove sugar resources

### PR DESCRIPTION
**Pull request by pierDipi**
- Remove deployment
- Remove service for monitoring
- Remove service monitor

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

Fixes JIRA #SRVKE-1193

### CODE REVIEW

File: knative-operator/pkg/controller/knativeeventing/knativeeventing_controller_test.go [hunk 1]
```diff
@@ -45,105 +42,3 @@ func init() {
 
 	apis.AddToScheme(scheme.Scheme)
 }
-
-// TestEventingReconcile runs Reconcile to verify if eventing resources are created/deleted.
-func TestEventingReconcile(t *testing.T) {
```
Comment (pierDipi) [inline knative-operator/pkg/controller/knativeeventing/knativeeventing_controller_test.go:50]: I converted this test to an E2E test, see VerifyDashboards.

**Comment by pierDipi** [comment]
`Validate / Generated files are commited (pull_request)` is failing because serving tests are failing on https://github.com/openshift-knative/serverless-operator/pull/1668

File: knative-operator/pkg/controller/knativeeventing/knativeeventing_controller_test.go (hunk 1 above)
Comment (matzew) [reply knative-operator/pkg/controller/knativeeventing/knativeeventing_controller_test.go:50]: +1

File: test/e2e/knative_eventing_test.go [hunk 2]
```diff
@@ -53,10 +53,18 @@ func TestKnativeEventing(t *testing.T) {
 		VerifyNoDisallowedImageReference(t, caCtx, eventingNamespace)
 	})
 
+	t.Run("Verify sugar controller deletion", func(t *testing.T) {
+		if err := test.CheckNoDeployment(caCtx.Clients.Kube, eventingNamespace, "sugar-controller"); err != nil {
```
Comment (matzew) [inline test/e2e/knative_eventing_test.go:57]: I think we can remove this in a future release.

File: test/upgrade/verify_upgrade.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package upgrade
+
+import (
+	"knative.dev/pkg/test/upgrade"
+
+	"github.com/openshift-knative/serverless-operator/test"
+)
+
+func VerifySugarControllerDeletion(ctx *test.Context) upgrade.Operation {
```
Comment (matzew) [inline test/upgrade/verify_upgrade.go:9]: likewise, to be removed in future release

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by pierDipi** [comment]
@matzew fixed the 2 GH actions jobs otherwise Prow won't merge the PR

**Comment by matzew** [comment]
> @matzew fixed the 2 GH actions jobs otherwise Prow won't merge the PR

force merge, FTW :sweat_smile:

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**